e2e/pages/app/badge/badge.testpage.component.html

```diff
@@ -0,0 +1,92 @@
+                <li>
+                    <ux-icon name="list" uxBadge badgeColor="partition9" [badgeVerticalPosition]="vertPos"
+                        [badgeHorizontalPosition]="horzPos" [badgeOverlap]="true" badgeSize="small"></ux-icon>
+                </li>
+            </ng-container>
+        </ng-container>
+    </ul>
+    <hr />
+    <!-- size examples - using icons -->
+    <ul class="element-list">
+        <li *ngFor="let size of sizes">
+            <ux-icon name="mail" uxBadge="NEW" badgeVerticalPosition="above" badgeHorizontalPosition="after"
+                [badgeSize]="size" badgeColor="critical"></ux-icon>
+        </li>
+    </ul>
+    <hr />
+    <!-- positioning examples - using buttons -->
+    <ul class="element-list">
+        <ng-container *ngFor="let vertPos of verticalPositions">
+            <ng-container *ngFor="let horzPos of horizontalPositions">
+                <li>
+                    <a href="#/badge" uxBadge="2" [badgeVerticalPosition]="vertPos"
+                        [badgeHorizontalPosition]="horzPos" badgeSize="medium" badgeColor="ok">
+                        Tasks
+                    </a>
+                </li>
+            </ng-container>
+        </ng-container>
+    </ul>
+    <hr />
+    <!-- overlap with positioning examples -->
+    <ul class="element-list">
+        <ng-container *ngFor="let vertPos of verticalPositions">
+            <ng-container *ngFor="let horzPos of horizontalPositions">
+                <li>
+                    <button class="btn btn-primary" uxBadge="2" [badgeVerticalPosition]="vertPos"
+                        [badgeHorizontalPosition]="horzPos" [badgeOverlap]="true" badgeSize="medium"
+                        badgeColor="chart2">
+                        Messages
+                    </button>
+                </li>
+            </ng-container>
+        </ng-container>
+    </ul>
+    <hr />
+    <!-- truncation examples -->
+    <ul class="element-list">
+        <li>
+            <button class="btn btn-primary" uxBadge="7483" badgeMaxValue="99" badgeVerticalPosition="above"
+                badgeHorizontalPosition="after" badgeSize="medium" badgeColor="chart4">
+                Alerts
+            </button>
+        </li>
+        <li>
+            <button class="btn btn-primary" uxBadge="Action required on items" badgeMaxValue="15"
+                badgeVerticalPosition="above" badgeHorizontalPosition="after" badgeSize="medium" badgeColor="chart4">
+                Alerts
+            </button>
+        </li>
+    </ul>
+    <hr />
+    <!-- hidden example -->
+    <a href="#/badge" uxBadge="7483" badgeMaxValue="99" badgeVerticalPosition="above" badgeHorizontalPosition="after"
+        badgeSize="medium" badgeColor="chart4" [badgeHidden]="true">
+        Hidden badge example
+    </a>
+    <hr />
+    <!-- color and border examples -->
+    <ul class="element-list">
+         <li *ngFor="let colors of colorCombinations.entries()">
```

Comment:
We should use the key value pipe here instead of calling the entries function:

https://angular.io/api/common/KeyValuePipe

Reply:
I didn't know about the K -> V pipe, nice!